x-pack/legacy/plugins/maps/index.js

```diff
@@ -108,111 +102,34 @@ export function maps(kibana) {
-            initRoutes(server, license.uid);
-          }
-        });
-
-      xpackMainPlugin.info
-        .feature(this.id)
-        .registerLicenseCheckResultsGenerator(checkLicense);
-
-      const sampleDataLinkLabel = i18n.translate('xpack.maps.sampleDataLinkLabel', {
-        defaultMessage: 'Map'
-      });
-      server.addSavedObjectsToSampleDataset('ecommerce', getEcommerceSavedObjects());
-      server.addAppLinksToSampleDataset('ecommerce', [
-        {
-          path: createMapPath('2c9c1f60-1909-11e9-919b-ffe5949a18d2'),
-          label: sampleDataLinkLabel,
-          icon: APP_ICON
-        }
-      ]);
-      server.replacePanelInSampleDatasetDashboard({
-        sampleDataId: 'ecommerce',
-        dashboardId: '722b74f0-b882-11e8-a6d9-e546fe2bba5f',
-        oldEmbeddableId: '9c6f83f0-bb4d-11e8-9c84-77068524bcab',
-        embeddableId: '2c9c1f60-1909-11e9-919b-ffe5949a18d2',
-        embeddableType: 'map',
-        embeddableConfig: {
-          isLayerTOCOpen: false
-        },
-      });
-
-      server.addSavedObjectsToSampleDataset('flights', getFlightsSavedObjects());
-      server.addAppLinksToSampleDataset('flights', [
-        {
-          path: createMapPath('5dd88580-1906-11e9-919b-ffe5949a18d2'),
-          label: sampleDataLinkLabel,
-          icon: APP_ICON
-        }
-      ]);
-      server.replacePanelInSampleDatasetDashboard({
-        sampleDataId: 'flights',
-        dashboardId: '7adfa750-4c81-11e8-b3d7-01146121b73d',
-        oldEmbeddableId: '334084f0-52fd-11e8-a160-89cc2ad9e8e2',
-        embeddableId: '5dd88580-1906-11e9-919b-ffe5949a18d2',
-        embeddableType: MAP_SAVED_OBJECT_TYPE,
-        embeddableConfig: {
-          isLayerTOCOpen: true
+      // legacy dependencies
+      const __LEGACY = {
+        config: server.config,
+        mapConfig() { return server.config().get('map'); },
+        route: server.route.bind(server),
+        plugins: {
+          elasticsearch: server.plugins.elasticsearch,
+          xpackMainPlugin: server.plugins.xpack_main
         },
-      });
-
-      server.addSavedObjectsToSampleDataset('logs', getWebLogsSavedObjects());
-      server.addAppLinksToSampleDataset('logs', [
-        {
-          path: createMapPath('de71f4f0-1902-11e9-919b-ffe5949a18d2'),
-          label: sampleDataLinkLabel,
-          icon: APP_ICON
-        }
-      ]);
-      server.replacePanelInSampleDatasetDashboard({
-        sampleDataId: 'logs',
-        dashboardId: 'edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b',
-        oldEmbeddableId: '06cf9c40-9ee8-11e7-8711-e7a007dcef99',
-        embeddableId: 'de71f4f0-1902-11e9-919b-ffe5949a18d2',
-        embeddableType: MAP_SAVED_OBJECT_TYPE,
-        embeddableConfig: {
-          isLayerTOCOpen: false
+        savedObjects: {
+          getSavedObjectsRepository: server.savedObjects.getSavedObjectsRepository
         },
-      });
+        addSavedObjectsToSampleDataset: server.addSavedObjectsToSampleDataset,
```

Comment:
These sample data-related methods are fine for now. They will be updated as part of https://github.com/elastic/kibana/pull/51126/files.